### DISCUSSION TITLE
[PROPOSAL] Cows 🐮 🤠

### DISCUSSION THREAD

**Discussion by spencerdezartsmith** [sunflower-land/sunflower-land#2240] [Ideas]
# Introducing Cows into Sunflower Land 🐄 

![wencows](https://user-images.githubusercontent.com/17863697/218600796-d3781280-c86c-4718-b20d-c5089c25cbce.png)

_This proposal is a live document. Changes may be made based on player feedback or based on changing requirements_

## Introduction

I am excited to introduce our proposal for bringing cows to Sunflower Land! This new addition will bring a whole new level of excitement and variety to the game. With the introduction of cows, players will have the opportunity to experience seasonal breeding, raising, and caring for their very own virtual herd.

Not only will players have the satisfaction of watching their herds grow, but they will also be able to collect and utilise the resources that come with having a cattle herd. Fresh milk and juicy steaks will be available for players to collect and use in a variety of ways, including cooking up delicious meals in the new steakhouse, processing milk into dairy products at the creamery and even selling cows for meat at the butcher.

I believe that the introduction of cows will bring the “next level” of animals to Sunflower Land 🌻. This game loop will require higher investment but will reap larger rewards. I believe this will enhance the overall gaming experience and provide players with new challenges that hopefully can be enjoyed many times over.

I look forward to hearing your thoughts on this proposal and can't wait to see these beautiful creatures on our lands.

_Please note that no values have been finalised. The values you see here are placeholders/rough ideas at this time._

## Getting started with Cows

To get started with cows you will first need to get one! There are multiple pathways you can take when it comes to cows. You may want to be a dairy farmer and just focus on getting a high-production herd that can keep the white gold flowing so you can eat ice cream and cheesecake with the hungry Goblins in the Grubshop. Maybe you’re lactose intolerant and it’s the 10-gallon hat that floats your boat. For you, the focus might be raising cattle for beef, focusing your attention on high-quality meat products, and eating out at the Steakhouse. For the risk-takers and strategic thinkers there is the seasonal breeding game, where each season you can try your luck at breeding for the Legendary Cow. This cow brings high rewards but isn’t an easy achievement. There will only be a small number of farmers who will get this prize of all prizes but those that miss out can clear out their herds and start again next season or just hang up their boots and enjoy the boosts that came with the lower tier cows they managed to get.

Below we will look at some of the core concepts around all of the elements that will come into play with this new addition to Sunflower Land.

### Barn 🐄 

Building Capacity: 8
Common Cow: 50 SFL
Common Bull: 75 SFL
Cost: TBD
Level: TBD

The Barn would be the place where you manage your herd. Here you can purchase a common cow or a common bull. It will also be where you "house" your animals.

When you purchase a cow from the Barn and place it on your land it will be ready to start producing milk immediately just need to get some wheat in her belly. A bull will only be required if you want to start breeding cattle.

## Animal Care ❤️ 

Caring for your cattle will take time, resources, and commitment. Each stage of a cow's life has different requirements. The table below outlines these:

![Screen Shot 2023-02-14 at 10 43 40 am](https://user-images.githubusercontent.com/17863697/218599540-df0e0fd9-458a-463d-adf9-b76bbf20bd5a.png)

Currently, the bull is kind of a "freeloader". You don't need to feed them BUT they do take up precious space in your Barn. 

There are different care requirements at different life stages. 

A cow requires wheat to produce milk. If you miss feeding her for a period of time her milk production will stop and she will become sad 😥. The only way to start her production again is to feed her a special diet of fruits along with her regular wheat. Keeping her fed will keep the milk flowing.

A pregnant cow needs to be fed 50% more wheat every 24hrs in order to keep the pregnancy going. If you miss a day that the calf will sadly be lost. You can either wait breed cooldown and then try again or lift her spirits with a healthy diet of fruits to get her to produce milk again.

A calf will need to drink milk to get big and strong. They will need to be fed every 24hrs so remember to have a little milk reserved for this ahead of the birth 🍼. If you miss a feed another day will be added to the time it takes for them to grow into a cow. Once they’re grown, they will start producing milk.

## Breeding

The breeding of cows will happen during a "season". This will almost be like a game within a game. The main goal of the breeding season will be to get that season's Legendary Cow. The Legendary cow will have significant boosts and will also have a unique design that will be different each season. The Legendary Cow will be the only one that will be withdrawable. In order to get a Legendary you will encounter some other tiers of cattle. These will also come with their own boosts which may or may not change with the seasons. 

Once the season ends so will all the breeding of that season's cattle. In effect, you will need to “start over” if you wish to try for the next season’s Legendary Cow. This will mean you need to clear out some space in your Barn to house the next season's additions. The end of the season will be an exciting time at the Butcher as there will be “Special Deals” offered for some of the prize cattle. These deals could potentially offer more than just larger hauls of meat.

**Note: Breeding is not something you need to do or something you have to continue doing. You may end up with a nice combination of rare, boosted cattle that you’re happy to enjoy for a while. It is totally up to you.**

There will be 5 tiers of cattle: 

- Common (not seasonal)
- Uncommon
- Rare
- Epic
- Legendary (cannot breed)

Breeding pairs of these together have different odds of producing the next tier of cow/bull. A cow can be bred multiple times however she will have a 3-day cooldown period between tries. Every time you breed you also run the risk of the breed not working.

_Note: Breeding from tiers that aren’t connected will produce offspring of the lower of the two tiers. Eg. Epic/Common would produce a Common only._

<img width="460" alt="Screen Shot 2023-02-13 at 9 03 14 am" src="https://user-images.githubusercontent.com/17863697/218600447-94a2d027-5094-4e28-8ca3-f7f7e21343fb.png">
_Note: These are not final numbers_

Each tier of cattle will offer different boosts. Below is an example of what the boosts might look like.

![Screen Shot 2023-02-14 at 10 52 21 am](https://user-images.githubusercontent.com/17863697/218600672-aedf3497-97e0-43a2-9cf1-d33b94b1ff9c.png)

_Note: These are not final numbers_

Breeding Requirements: 

Wheat: 30
Apple: 20
SFL: 10

Notes of breeding: 

There is also an ongoing discussion about where Legendary's cows fit within your Barn. As mentioned above, space is tight. Having a Legendary take up space can limit your ability to breed for another Legendary as your space is now reduced by 1. 

Some ideas:

- Legendary increases capacity by 1
- Special 5-star Barn to house Legendary’s
- Do nothing

## Buildings

Due to cows being a much more complex and expensive system, foods produced using the resources from these animals will have a great xp gain attached to them. The Grub Shop rewards will also reflect the hard work players have put into this mechanic.

We are waiting on finalising the values of the Cow Ecosystem before finalising values for the foods they produce. Therefore, there will be no numbers attached to the buildings/foods listed below.

### Butcher 🥩 

The butcher is where you will sell your cattle for meat. Different tiers of cows will fetch different amounts of meat at the butcher. As I mentioned above, the end of a breeding season will be high activity at the butcher as you will be able to fetch some good deals for the cattle you no longer wish to keep. 

You will then be able to use the meat in recipes at the Steakhouse.

### Steakhouse 🍔 

Here you will be able to make delicious meat-based recipes for your bumpkins such as:

- Hamburger
- Cheeseburger
- Steak and potatoes
- Cured meat and cheese plate
- Steak and veggies

### Creamery 🧀 

Here you be able to utilise your milk to make other dairy products such as:

- Fancy cheeses
- Yoghurt
- Ice Cream
- Milkshakes
- Cheesecake

#### The "Young Cow" (Hiefer)

In the real world, a cow can only start to produce milk once she has given birth. This doesn't necessarily have to happen in Sunflower Land though. If we didn't do this, then once the calf has grown (3 days) she would be able to be fed and produce milk.

#### Cow Mortality

Cows need to be fed. Just as a calf can be lost, a cow could also pass over if not fed for X amount of days.

#### Cow reproduction not being so “lucky”

Genetics is something that would be amazing to implement and maybe could be introduced further down the line but to start with we could have cows that are just better at producing due to other factors such as high-quality food or “treats” of some sort. 

#### Add-ons to current foods

Having the ability to add things like cheese, and sour cream to other foods to boost the XP they offer.

### Cow-based Skills

As we introduce more animals we can start to play around with more focused skills that relate to each animal. These are already getting drafted at the moment. Keep your eyes on the proposals that are currently out there.

### Bumpkin Wearables

There are a lot of cool wearable opportunities for our Bumpkins when it comes to cows. Think… hats, boots, whips… handbags??? 😜

![The-return-of-Pitti-Uomo-59038-detailp](https://user-images.githubusercontent.com/17863697/218600956-43fd99e6-a4a1-41cf-a7ca-32b0a577bb45.jpeg)

## Final Thoughts

There are many numbers that need to be balanced when it comes to introducing cows. The idea with this is to make it much more dynamic than chickens. This will be a more complex mechanic that will be aimed at the higher levels of the game. Introducing the seasonal approach brings added excitement that allows players to jump in and out of the process at their leisure. It also allows for things to be tweaked and changed every 3 months. 

The Legendary Cows are something we want to make very special. Their rarity and unique designs will make the sort-after collectibles but their boosts should also be impactful. We’re still playing around with balancing what these boosts could be. I would love to hear some of your thoughts on this. 

❤️  Massive thanks to TheDcol for being my brainstorming partner on this proposal.

**Comment by ShiningMinions** [comment]
I think overall this is looking great! I would definitely be most interested in breeding, I love going for those rare items :smile: .
Here are some thoughts to consider:

- Mutant cows: I think they should be incorporated somehow. We have mutant chickens, and there are many boosts increasing mutant chance for animals.

- I definitely like the more-complex mechanics, but I would suspect we need to nail this UI so it doesn't become over-complicated for the player. For example, I can already see players complaining in farmer's chat "not knowing they needed to feed their pregnant cow", and now they've "lost their resources", etc. UI/UX will be extremely important IMO. 

- Without going too far out-of-scope: For the "butcher", I would be interested in seeing chickens somehow going to the butcher shop as well. Or even better, perhaps we can find a way to butcher mutant chickens, as that would provide a much-needed burn.

**Reply by spencerdezartsmith** [reply]
Oooooh butchering the mutants would be so good. That's the one thing I don't like about them is that they just keep going forever... 

I definitely think there would be space to get chickens into the butcher somehow once it's introduced.

Mutant cows are a nice idea as well. It would be cool if there was a chance they could show up during any breed ie. common x common. 

The UX will be key in this mechanic. We can try to utilise icons similar to how the sims use them to communicate needs of characters. Since a lot of things are time sensitive we will need to have this nailed down.

**Reply by TheDcol** [reply]
Thanks Shining for the great comment!

- We were thinking of the rarer cows to act as mutants of sorts, mutant chickens act as a boost to all chickens whilst the rarer cow boosts itself. We can have a deeper think, but would be tricky imo to differentiate between mutant cow and an epic/legendary cow. Although, as a wrangler farmer, i would want to leverage that and be able to have an edge over others, so there is possibility there to include mutants 😁

- Yes, the UI needs to be on point 💪

- I think we can work something out too for those chickens 😏 but depends if devs want to expand that aspect too i guess

**Reply by rasheed19871** [reply]
i love new mechanics in sunflower chicken, cows fishing name it, one thing i like the most is having so many things to do in sunflower land 24/7. i love sunflower team

**Comment by SarielWang93** [comment]
I wonder if golden rooster has an effect on cows eating wheat.

**Reply by spencerdezartsmith** [reply]
Not likely. The rooster affecting the chickens makes sense but not so much for cows. I think there is space for there to be solid boosts coming from the Legendary Cows though. 🤔

**Comment by paduv** [comment]
Sounds great! I like the breeding idea, can't wait for this!

The butcher thing might be a bit divisive though, it's gonna be hard making that look 100% cute..
Maybe Cows could be replaced by some monstruous magical animal? That would free us from any potential ethical issues.
Or maybe add an alternative to butchering, so that vegans can play the game without feeling bad.
I don't know, maybe i'm too cautious & nobody cares about that :)

**Reply by TheDcol** [reply]
Definitely came across my mind for sure, I think not including anything too graphic/blood might be the way to go.. 

You could trade in the animals and the goblin butchers would simply give you packs of meat..  my partner is vegetarian so might ask her what she thinks too 😅 

Valid concern! 😁

**Reply by spencerdezartsmith** [reply]
This is definitely something that came across my mind too. Maybe we could trade the animal to the goblin butcher and then you get back impossible meat (fake meat). There can be some narrative that goblins are vegetarians and they put on the cows on cow island where they live happily forever.

**Reply by TheDcol** [reply]
> This is definitely something that came across my mind too. Maybe we could trade the animal to the goblin butcher and then you get back impossible meat (fake meat). There can be some narrative that goblins are vegetarians and they put on the cows on cow island where they live happily forever.

Ohhh not bad! Build some lore around goblins collecting cows for a secret project they are planning.... and they trade in this magic meat to the player... Players can decide themselves if goblins are really butchering cows or planning something else 👀

**Reply by ShiningMinions** [reply]
> Definitely came across my mind for sure, I think not including anything too graphic/blood might be the way to go.. You could trade in the animals and the goblin butchers would simply give you packs of meat..

@TheDcol  Agreed :+1: - I also think calling it a "meat market" instead of a "butcher" might be a good option. Sounds less intense for more sensitive players.

> Maybe we could trade the animal to the goblin butcher and then you get back impossible meat (fake meat). There can be some narrative that goblins are vegetarians and they put on the cows on cow island where they live happily forever.

@spencerdezartsmith  To be honest, I'm not a fan of this. I understand the need to be sensitive to particular players/cultures, but there's a point where it's too much and it feels "lame" to players like me. I think simply trading animals at the "meat market" to receive "meat packs" or "meat food" is good.

**Reply by Alaindrea** [reply]
my partner really likes the idea of the goblins handing you a package of  "meat?" lol He literally spoke the question mark. 
As someone who played a goblin in the war I think it is equally plausible that goblins eat the cows, worship the cows, or maybe can speak to them and use them as spies to see if the moon seekers have left the mainland yet. 

I also think it would be appropriate for players to have another option besides feeding their bumpkin meat/trading the cows to the "meat market" such as selling the "live" cow to another player.

On a side note, I think it is interesting that more people are having an issue with the possibly acknowledging real cows get eaten than the subtle thing about cow miscarriages. Jus' saying. It is a lot easier to explain to a kid that people eat animals to make meat than to explain miscarriage, and it would be really bad if some young woman starved herself hoping to miscarry an unwanted pregnancy because she saw it on a game.

Honestly, I like both mechanics. However, I'd say that it might be better to say that if you don't feed the pregnant cow extra every day in the game, then you get a skinny calf which gives only 1/2 the milk or meat when it grows up.

**Comment by IvanSkazkin** [comment]
Perhaps a golden rooster could reduce the amount of wheat needed for cows by 20%?

It wouldn't hurt as much, but it would have an interesting meaning. After all, logically. The rooster from somewhere will give wheat for chickens, but he cannot get wheat for cows. And after all, if we have 20 chickens on the farm, he knows where to get 20 wheat. If we put up a chicken coop and we increase the number of hens, the golden rooster still knows how to get the wheat. So I think it would be reasonable.

I just don't see where else eggs are needed.

I'm level 40 already. I hope I can get not only experience on cows, but also earnings sfl, make a separate one more grub shop for cows

**Reply by TheDcol** [reply]
Thanks for the feedback,

I think golden rooster should have nothing to do with cows, it inspires the chickens to eat without being fed. I am saying this as someone who HAS golden rooster btw 😁

Regarding your concern with eggs/food in general... I see it too:

- Why should i make more food? I am at max level..

A few things to keep in mind... I view level 40 as just the beginning. The following proposal for revamped skill trees presents itself as an opportunity to introduce requirements of 168 skill points to start with, this is without other skill trees being brainstormed on for the future such as fishing, coffee/tea etc: 

https://github.com/sunflower-land/sunflower-land/discussions/2180

1 skill point every 1.5 levels = 245~ levels to max all of those out for 1 bumpkin. Food will be very important if that goes ahead 😁

There is also a multiple bumpkin proposal which might encourage more food being consumed: https://github.com/sunflower-land/sunflower-land/discussions/2211

There are ALSO more foods coming out with eggs too if you check the other food art proposal thread.

In summary, I agree that food needs to be important and be more than what it is now, i believe milk/steaks will yield players with great xp/sfl (grub shop) and be very important if you want to unlock more and more skills

Hope that was enough info, btw with cows needing wheat it means there should in theory be less wheat circulated which means less supply of it for chickens... Spitballing here but higher wheat market price usually tends to higher egg price 💪

**Comment by aimlemonade** [comment]
Uhmm I think the rarity part is a bit random, does not make much sense for the game! Excluding the cows having rarity everything else seems good to me!

**Comment by Blasta-alasta** [comment]
Hey Spencer, I love the proposal... especially the complexity and strategy that that looks to enhance gameplay for more advanced players! I also put together a Cows proposal of sorts a few days ago (not sure if you saw), and would love for you to check it out when you have time. I feel like some some of ideas might be able to be incorporated into this proposal, if not, thats all good too. 😅

It's here https://github.com/sunflower-land/sunflower-land/discussions/2227

**Reply by spencerdezartsmith** [reply]
I did see it! I haven't had a chance to give it a good read yet but I will for sure!

**Comment by Xeko-92** [comment]
>Cow Mortality
Cows need to be fed. Just as a calf can be lost, a cow could also pass over if not fed for X amount of days.

Don't love this. Imagine your legendary cows died because you went on vacation. While I think I'd still be able to maintain the cows if I was on vacation, I'd just hate if this happened to someone because of a natural disaster or something. Seems unnecessarily punishing if we already have a built in sad cows thing.

Also a question. In theory, do you just want 1 bull, ideally legendary? And then you can use that to breed with all the cows.

**Reply by spencerdezartsmith** [reply]
I actually think there should only be a Legendary Cow no Bull. Reason being is that the Buill is really only useful for breeding and there is no breeding at the Legendary level. Reaching a LC is like the end of the breeding game. You won. 

Yeah, I agree with the cow mortality issue you raised. A little too high maintenance for a casual farming game 😅

**Reply by InuBakaBo** [reply]
Maybe a reduced productivity of milk from the hungry cows instead of death

**Reply by Xeko-92** [reply]
Is there any reason to have more than 1 bull? The meta would be to have 1 highest tier bull as possible and then 7 cows right?

**Reply by TheDcol** [reply]
> Is there any reason to have more than 1 bull? The meta would be to have 1 highest tier bull as possible and then 7 cows right?

You need an epic bull to breed with epic cow to have a chance to produce a legendary cow

Breeding a common bull with an epic cow reverts the chances at common level

Not final, but this is the general thought

**Reply by brian-man** [reply]
> You need an epic bull to breed with epic cow to have a chance to produce a legendary cow
> 
> Breeding a common bull with an epic cow reverts the chances at common level

Setting aside the percentages for a moment, I think it should be expressed in math terms.  Elaborating...

- 1 Common (not seasonal)
- 2 Uncommon
- 4 Rare
- 8 Epic
- 16 Legendary (cannot breed)

Common+Common = 1+1 = 2 so best result is Uncommon
Uncommon+Uncommon = 2+2 = 4 so best result is Rare
Rare+Rare = 4+4 = 8 so best result is Epic
Epic+Epic = 8+8 = 16 so best result is Legendary

Common+Epic = 1+8 = 9 so best result is another Epic
...
Rare+Epic = 4+8 = 12 so best result is another Epic

Now you can further use those numbers to adjust the percentages as well.  For example, R+R=8 vs C+E=9 vs R+E=12 so the last case (R+E) is more likely to roll that Epic than the other two cases where the additive value of the parents is in the range [8,15].

Note that addition is just one option.  You could use multiplication.  And of course you could choose numerical values for the rarity levels that are something other than binary digit weights.

An additional benefit of the numeral approach is that the code is easier to write, debug, verify, etc. Instead of a chain of if statements where the comments are bigger than the code, you just have a simple math formula based on the values assigned to the parent rarities.

**Reply by TheDcol** [reply]
A good alternative! More elegant too

**Reply by Alaindrea** [reply]
> Maybe a reduced productivity of milk from the hungry cows instead of death

I'd say hungry cows (those that have not been fed in the last 24 hours) just don't give milk at all.

**Comment by sergeniy1** [comment]
top photo, it is necessary to take a photo as a team))))
please tell me where to write, there is an idea for all farmers how to use a 2x2 bed, where you can place it with a potion once in a hole under a fruit tree

**Comment by Chucrutty** [comment]
> > This is definitely something that came across my mind too. Maybe we could trade the animal to the goblin butcher and then you get back impossible meat (fake meat). There can be some narrative that goblins are vegetarians and they put on the cows on cow island where they live happily forever.
> take of word meat, butcher and we will be fine i think

> Ohhh not bad! Build some lore around goblins collecting cows for a secret project they are planning.... and they trade in this magic meat to the player... Players can decide themselves if goblins are really butchering cows or planning something else 👀

**Comment by ispankzombiez** [comment]
frankly. i love it. good work friend :)

**Comment by Munsie94** [comment]
I really like this proposal
